libcxx/include/__tree

```diff
@@ -1007,28 +1056,38 @@ public:
   }
 
   template <class _ValueT = _Tp, __enable_if_t<__is_tree_value_type_v<_ValueT>, int> = 0>
-  _LIBCPP_HIDE_FROM_ABI void
+  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX26 void
   __insert_unique_from_orphaned_node(const_iterator __p, __get_node_value_type_t<_Tp>&& __value) {
-    __emplace_hint_unique(__p, const_cast<key_type&&>(__value.first), std::move(__value.second));
+#if _LIBCPP_STD_VER >= 26
+    if constexpr (integral_constant<bool, is_copy_constructible<decltype(__value.first)>::value >()) {
```

Comment:
Can't we just use `if constexpr(is_copy_constructible_v<...>`? Also, let's only use this branch in constant evaluation.